.github/workflows/check.yml

```diff
@@ -177,6 +177,22 @@ jobs:
                   components: rust-src
                   default: false
 
+            - name: Cache cargo index
+              uses: actions/cache@v2.1.4
+              with:
+                  path: ~/.cargo/registry/index # https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci
+                  key: cache-cargo-index-${{ github.run_id }} # https://github.com/actions/cache/issues/342#issuecomment-673371329
+                  restore-keys: cache-cargo-index-
```

Comment:
Are you sure that this index is os-independent? Maybe it's worth just merging both `Cache cargo *` tasks into single one and create separate cache for each os?